lib/opset.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {Graph} from './graph';
+import {Operator} from './operators';
+
+export interface OpSet {
+  domain: string;
+  version: number;
+}
+
+export declare namespace OpSet {
+  interface OperatorConstructor {
+    (node: Graph.Node): Operator;
+  }
+
+  /**
+   * Domain of an opset, it can be an empty string(default value, represent for ai.onnx), or 'ai.onnx.ml'
+   */
+  type Domain = ''|'ai.onnx.ml';
+
+  /**
+   * A resolve rule consists of 4 items: opType, opSetDomain, versionSelector and operatorConstructor
+   */
+  type ResolveRule = [string, Domain, string, OperatorConstructor];
+}
+
+export function resolveOperator(
+    node: Graph.Node, opsets: ReadonlyArray<OpSet>, rules: ReadonlyArray<OpSet.ResolveRule>) {
+  for (const rule of rules) {
+    const opType = rule[0];
+    const domain = rule[1];
+    const versionSelector = rule[2];
+    const opConstructor = rule[3];
+
+    if (node.opType === opType) {  // operator type matches
+      for (const opset of opsets) {
+        // opset '' and 'ai.onnx' are considered the same.
+        if (opset.domain === domain || (opset.domain === 'ai.onnx' && domain === '')) {  // opset domain found
+          if (matchSelector(opset.version, versionSelector)) {
+            return opConstructor(node);
+          }
+        }
+      }
+      throw new TypeError(`cannot resolve operator '${opType}'`);
```

Comment:
can error thrown also mention requested opset version of the operator cannot be resolved ?